src/Text/Pandoc/Parsing.hs

```diff
@@ -1468,12 +1468,20 @@ citeKey = try $ do
   guard =<< notAfterString
   suppress_author <- option False (True <$ char '-')
   char '@'
-  firstChar <- alphaNum <|> char '_' <|> char '*' -- @* for wildcard in nocite
-  let regchar = satisfy (\c -> isAlphaNum c || c == '_')
-  let internal p = try $ p <* lookAhead regchar
-  rest <- many $ regchar <|> internal (oneOf ":.#$%&-+?<>~/") <|>
-                 try (oneOf ":/" <* lookAhead (char '/'))
-  let key = firstChar:rest
+  key <-
+    do firstChar <- alphaNum <|> char '_' <|> char '*' -- @* for wildcard in nocite
+       let regchar = satisfy (\c -> isAlphaNum c || c == '_')
+       let internal p = try $ p <* lookAhead regchar
+       rest <- many $ regchar <|> internal (oneOf ":.#$%&-+?<>~/") <|>
+                      try (oneOf ":/" <* lookAhead (char '/'))
+       pure $ firstChar:rest
+    <|> char '{' *>
+        many `id` choice
+          [ noneOf "}\n\\"
+          , ' ' <$ newline <* skipSpaces <* notFollowedBy newline
+          , char '\\' *> anyChar
+          ]
+        <* char '}'
```

Comment:
> As for newlines, I think it would make sense to allow newlines only if not followed by a blankline (there's a parser for this

I think you mean `spnl`? That one is in `Readers/Markdown.hs`, so not available here unless we move it.

Reply:
I wondered if it would be more consistent to parse any occurrence of multiple spaces as a single space. The way it is now I would think it may be especially confusing with trailing spaces? If one wants multiple spaces they can still escape them.

Reply:
Yes, we could collapse consecutive spaces, that might make sense.